src/OcisResource.php

```diff
@@ -558,4 +558,31 @@ private function getFileResponseInterface(string $fileId): ResponseInterface
         $webDavClient->setCustomSetting($this->connectionConfig, $this->accessToken);
         return $webDavClient->sendRequest("GET", $fileId);
     }
+
+    /**
+     * @throws UnauthorizedException
+     * @throws ForbiddenException
+     * @throws BadRequestException
+     * @throws NotFoundException
+     * @throws InternalServerErrorException
+     * @throws InvalidResponseException
+     * @throws HttpException
+     */
+    public function getPreview(string $width, string $height): string
+    {
+        $webDavClient = new WebDavClient(['baseUri' => $this->serviceUrl . '/dav/spaces/']);
+        $webDavClient->setCustomSetting($this->connectionConfig, $this->accessToken);
```

Comment:
we are keep on repeating this two lines. Shouldn't we use `Drive::createWebDavClient` ?

Reply:
adding a function in OcisResource is also fine